packages/wonder-blocks-icon-button/src/components/node-icon-button.tsx

```diff
@@ -0,0 +1,316 @@
+    "--wb-node-icon-button--box-background": string;
+    "--wb-node-icon-button--box-shadow-color": string;
+    "--wb-node-icon-button--box-padding": string | number;
+    "--wb-node-icon-button--box-shadow-y-rest": string | number;
+    "--wb-node-icon-button--box-shadow-y-hover": string | number;
+    "--wb-node-icon-button--box-shadow-y-press": string | number;
+    "--wb-node-icon-button--icon-size": string | number;
+}>;
+
+type Props = Omit<BaseIconButtonProps, "kind" | "style"> & {
+    /**
+     * The action type of the button. This determines the visual style of
+     * the button. Defaults to `notStarted`.
+     *
+     * - `notStarted` is used for buttons that indicate a not started action.
+     * - `attempted` is used for buttons that indicate an attempted (in progress)
+     *   action.
+     * - `complete` is used for buttons that indicate a complete action.
+     */
+    actionType?: "notStarted" | "attempted" | "complete";
+    /**
+     * The alternative text for the icon button. Required for accessibility.
+     */
+    "aria-label": string;
+
+    /**
+     * The size of the icon button.
+     * One of `small` (24) or `large` (68).
+     * Defaults to `large`.
+     */
+    size?: "small" | "large";
+
+    /**
+     * Custom styles for the elements in the ActivityIconButton component.
+     * - `root`: Styles the root element (button)
+     * - `box`: Styles the "chonky" box element
+     * - `icon`: Styles the icon element
+     */
+    styles?: {
+        root?: StyleType;
+        box?: StyleType;
+        icon?: StyleType;
+    };
+};
+
+/**
+ * Node buttons are visual representations of activities along in a Learning
+ * Path. When a represented Node is a button that launches the activity. Nodes
+ * use the Chonky shadow style.
+ *
+ * ```tsx
+ * import magnifyingGlassIcon from
+ * "@phosphor-icons/core/regular/magnifying-glass.svg";
+ * import {NodeIconButton} from "@khanacademy/wonder-blocks-icon-button";
+ *
+ * <NodeIconButton
+ *     icon={magnifyingGlassIcon}
+ *     aria-label="An Icon"
+ *     onClick={(e) => console.log("Hello, world!")}
+ * />
+ * ```
+ */
+export const NodeIconButton: React.ForwardRefExoticComponent<
+    Props &
+        React.RefAttributes<typeof Link | HTMLButtonElement | HTMLAnchorElement>
+> = React.forwardRef<
+    typeof Link | HTMLButtonElement | HTMLAnchorElement,
+    Props
+>(function NodeIconButton(props: Props, ref) {
+    const {
+        actionType = "notStarted",
+        disabled = false,
+        icon,
+        size = "large",
+        styles: stylesProp,
+        type = "button",
+        // labeling
+        "aria-label": ariaLabel,
+        ...restProps
+    } = props;
+
+    const [pressed, setPressed] = React.useState(false);
+
+    const buttonStyles = [
+        styles.button,
+        disabled && styles.disabled,
+        !disabled && pressed && styles.pressed,
+        variants.size[size] as any,
+        variants.actionType[actionType] as any,
```

Comment:
Is there a way to add types to `variants` so that we don't need to cast it to `any` here?

Reply:
I modified this to use `StyleType`, so Aphrodite won't complain, but still an improvement over `any` :)